cpp/src/text/subword/bpe_tokenizer.cuh

```diff
@@ -30,30 +32,84 @@
 #include <cuco/static_map.cuh>
 
 #include <cstdint>
+#include <type_traits>
 
 namespace nvtext {
 namespace detail {
 
+using hash_value_type    = uint32_t;
+using string_hasher_type = cudf::hashing::detail::MurmurHash3_x86_32<cudf::string_view>;
+
+/**
+ * @brief Hasher function used for building and using the cuco static-map
+ *
+ * This takes advantage of heterogeneous lookup feature in cuco static-map which
+ * allows inserting with one type (index) and looking up with a different type (string).
```

Comment:
"index" seems like the wrong word here (and in the function signature). I'd propose "hash value" or similar. Indices are ordered, but hash values are just arbitrary integers.

Reply:
Ah, okay. My mistake.